code/Controllers/CMSMain.php

```diff
@@ -1997,7 +1998,7 @@ public function archive(array $data, Form $form): HTTPResponse
         if (!$record || !$record->exists()) {
             throw new HTTPResponse_Exception("Bad record ID #$id", 404);
         }
-        if (!$record->canArchive()) {
+        if (!Deprecation::withNoReplacement(fn() => $record->canArchive())) {
```

Comment:
```suggestion
        $canArchive = $record->canArchive();
        if (!canArchive)) {
```
Also update other instance in this PR

Reply:
Think I got them all.